### ISSUE TITLE
SLDF mechs not being migrated to ME modules correctly

### ISSUE THREAD

**Issue by PainGod85** [BattletechModders/MechEngineer#191]
List of mechs affected:
[log.txt](https://github.com/BattletechModders/MechEngineer/files/4906580/log.txt)

Zeus ZEU-5T (FF)
Crab CRB-27b (FF)
Warhammer WHM-7A (ES) <- this mech should have a 280 engine core, yet only receives a 250
Marauder MAD-2R (FF)
BattleMaster BLR-1Gb (ES)
Phoenix Hawk PXH-1b (gets a 190 engine core, should have a 270, missing XL, ES)
AS-7-D-HT (gets DHS, but engine core is a 285 when it should be a 300 with +2 internal sinks)
GRF-2N (engine core is a 260, should be a 275 with +1 sink - also missing FF, ES and - strictly speaking - ECM, but that's not part of your mod)

Basically, they get SHS, standard armor, structure and engines when all of them should have DHS, with some others getting various kinds of addon internal sinks, XL engines, ES structure or FF armor.

These are all the ones I can check easily now. Some of the common mechs are probably going to throw up some engine rating artifacts as well, but I can't easily check those without running across them in the campaign.

Case in point, the Flashpoint exclusive ANH-JH is running a 215 rated engine when it should have a 200.

**Comment by CptMoore** [comment]
a ecm prototype is part of the urban warfare dlc, so I could add that to the grf.. but maybe im blind, I dont see any GRF-2N.

**Comment by PainGod85** [comment]
That's because originally the mech was mislabeled as Griffin 4N. They eventually fixed the name to the correct version on the UI side, but internally they needed to retain the incorrect 4N designation for the sake of backwards compatibility.

**Comment by CptMoore** [comment]
ah damn, I have just pushed a commit that modifies the 2N instead of the 4N D:

**Comment by PainGod85** [comment]
Yeah, like I mentioned on Discord, that was me being overzealous since it was lacking FF. :(

**Comment by CptMoore** [comment]
ok, I had trouble for some mechs taking my changes, looks like with modloader (not modtek) DLC mechs can't be patched :(

**Comment by CptMoore** [comment]
Can you retest how the mechs behave? I made major imorovements to the autofixer. I would be allowed to add engine, ff and es to fix discrepancies still around
